### PR TITLE
[style] Improve fences and city walls rendering

### PR DESCRIPTION
**Pull request by aeshaganatra**
Create different color and style for fences and city wall.

### Fences
Z16
fencesZ16
Z17
fencesZ17
Z18
image

### City-Wall
Z16
cityWallZ16
Z17
CityWallZ17
Z18
image

Fixes: #2063

### CODE REVIEW

File: data/styles/clear/include/Basemap.mapcss [hunk 1]
```diff
@@ -670,34 +672,48 @@ area|z18-[building=train_station],
 /* 8.3 Barrier */
 
 line|z16-[barrier=wall],
-line|z16-[barrier=city_wall],
 line|z16-[barrier=retaining_wall],
 line|z16-[barrier=hedge],
-line|z16-[barrier=fence],
 line|z16-[historic=citywalls],
```
Comment (pastk) [inline data/styles/clear/include/Basemap.mapcss:677]: I think it makes sense to render all types of walls similarly?
And make only fences and hedges different.

File: data/styles/clear/include/Basemap.mapcss [hunk 2]
```diff
@@ -670,34 +672,48 @@ area|z18-[building=train_station],
...
+/* 8.4 Fence */
+ 
+line|z16-[barrier=fence],
+{color: @fencecolor; opacity: 0.5;}
+line|z16[barrier=fence],
+{width: 3.0; dashes: 2.7,2.7; opacity: 0.2;}
+line|z17-[barrier=fence],
+{width: 2.0; dashes: 2.0,2.0; opacity:0.3;}
+line|z18-[barrier=fence],
+{width: 1.0; dashes: 1.5,1.5; opacity: 0.6;}
```
Comment (pastk) [inline data/styles/clear/include/Basemap.mapcss:705]: Hmm, interesting.
Usually we increase width on higher zooms to make the feature look bigger.
What is the logic to do it vice versa?

File: data/styles/clear/style-clear/colors.mapcss [hunk 3]
```diff
@@ -78,7 +78,9 @@
 @aerodrome5: #C8BCDE;
 
 /* 4.3 Barriers */
-@fence: #999999;
+@barriers: #999999;
+@fencecolor: #000000;
+@citywallcolor: #732f2f;
```
Comment (pastk) [inline data/styles/clear/style-clear/colors.mapcss:83]: we don't add "color" to the color name, let's keep it consistent across the file

Comment (aeshaganatra) [reply data/styles/clear/style-clear/colors.mapcss:83]: make sense

File: data/styles/clear/include/Basemap.mapcss (hunk 2 above)
Comment (aeshaganatra) [reply data/styles/clear/include/Basemap.mapcss:705]: as per the requirement, 
biker was not able to see the fence on lower zoom level. 
so make it  high width on lower zoom level and make it small width at higher zoom level,
I was thinking in that way.

File: data/styles/clear/include/Basemap.mapcss (hunk 1 above)
Comment (aeshaganatra) [reply data/styles/clear/include/Basemap.mapcss:677]: sure, will update the PR

File: data/styles/clear/include/Basemap.mapcss (hunk 2 above)
Comment (pastk) [reply data/styles/clear/include/Basemap.mapcss:705]: We need to improve visibility on all zoom levels I think so let's keep it consistent with other line-type features rendering - the feature should get bigger the more we zoom in

**Review by pastk — CHANGES_REQUESTED** [review]
Your last push contains unrelated 3party/boost and twine submodules changes.
Please rebase onto a fresh master and `git submodule update --init`.

Also please remove `drules_proto_BACKUP*` and similar new files.

**Review by pastk — COMMENTED** [review]
Style-wise I think it could be like:
fence - thin black or dark gray line, solid or dotted
hedge - medium dark green solid line
walls - thick brown or brown-gray solid line

But need to see it on screenshots to decide (including night mode).
What do you think?

**Comment by pastk** [comment]
Done in #2756

**Pull request closed without merging**